knack/log.py

```diff
@@ -140,22 +130,31 @@ def configure(self, args):
             self._init_logfile_handlers(root_logger, cli_logger)
             get_logger(__name__).debug("File logging enabled - writing logs to '%s'.", self.log_dir)
 
-    def _determine_verbose_level(self, args):
+    def _determine_log_level(self, args):
         """ Get verbose level by reading the arguments. """
-        verbose_level = 0
-        for arg in args:
-            if arg == CLILogging.VERBOSE_FLAG:
-                verbose_level += 1
-            elif arg == CLILogging.DEBUG_FLAG:
-                verbose_level += 2
-        # Use max verbose level if too much verbosity specified.
-        return min(verbose_level, len(self.console_log_configs) - 1)
+        # arguments have higher precedence than config
+        if CLILogging.ONLY_SHOW_ERRORS_FLAG in args:
+            if CLILogging.DEBUG_FLAG in args or CLILogging.VERBOSE_FLAG in args:
+                raise CLIError("--only-show-errors can't be used together with --debug or --verbose")
+            self.cli_ctx.only_show_errors = True
+            return 1
+        if CLILogging.DEBUG_FLAG in args:
+            self.cli_ctx.only_show_errors = False
```

Comment:
>self.cli_ctx.only_show_errors = False [](start = 12, length = 37)

It seems not necessary, I think by default it is false

Reply:
This line is to make sure the config is overridden by `--debug`. If the config is `only_show_errors=True`, `--debug` will revert it to `False`.

Reply:
I see you has check debug cannot be set while only_show_errors set

---
In reply to: [394914515](https://github.com/microsoft/knack/pull/181#discussion_r394914515) [](ancestors = 394914515)

Reply:
The check at L136 is only to check the argument `--only-show-errors`. It doesn't raise error when the config has the `only_show_errors=True`. In this case, `self.cli_ctx.only_show_errors` is reverted by the `--debug` argument.